### PR TITLE
WTBUILD-111 Attach IAM role when launching instance when requested

### PR DESCRIPTION
**Pull request by etienneptl**

### CODE REVIEW

File: scripts/testy_launch.py [hunk 1]
```diff
@@ -191,6 +191,16 @@ def launch_from_distro(distro):
         instance_id = result.stdout.strip()
         wait_on_status_check(instance_id)
 
+        # Attach IAM role if requested.
+        if iam:
+            result = local(f"aws ec2 associate-iam-instance-profile \
+                --instance-id {instance_id} \
+                --iam-instance-profile Name=testy_iam", hide=True, warn=True)
```
Comment (ruby-oujo) [inline scripts/testy_launch.py:198]: do you think we should allow for choosing which IAM or no?

File: scripts/testy_launch.py [hunk 2]
```diff
@@ -191,6 +191,16 @@ def launch_from_distro(distro):
...
+            if result.stderr:
+                print("Failed at attaching IAM role.", flush=True)
+            else:
+                print("IAM role attached!", flush=True)
```
Comment (ruby-oujo) [inline scripts/testy_launch.py:202]: perhaps say the role name here testy_iam role attached?

**Review by ruby-oujo — APPROVED** [review]

File: scripts/testy_launch.py (hunk 1 above)
Comment (etienneptl) [reply scripts/testy_launch.py]: Yes, good idea. See 029545995482730f01632ca73d4de44f2f07c1b0

File: scripts/testy_launch.py (hunk 2 above)
Comment (etienneptl) [reply scripts/testy_launch.py]: I have updated the logs in 029545995482730f01632ca73d4de44f2f07c1b0.

**Review by ruby-oujo — APPROVED** [review]

**Pull request merged**